tests/e2e/fixtures/kserve_denpendencies.py

```diff
@@ -0,0 +1,196 @@
+        )
+        assert retcode == 0
+    else:
+        print("serving directory already exists, skipping clone ...")
+
+
+@pytest.fixture(scope="class")
+def associate_oidc(cluster, region):
+    associate_iam_oidc_provider(cluster_name=cluster, region=region)
+
+
+@pytest.fixture(scope="class")
+def kserve_iam_service_account(metadata, cluster, region, request):
+    metadata_key = "aws-sa"
+
+    service_account_name = "aws-sa"
+
+    def on_create():
+        create_iam_service_account(
+            service_account_name=service_account_name,
+            namespace="kubeflow-example-com",
+            cluster_name=cluster,
+            region=region,
+            iam_policy_arns=["arn:aws:iam::aws:policy/AmazonS3FullAccess"],
+        )
+
+    def on_delete():
+        delete_iam_service_account(
+            service_account_name=service_account_name,
+            namespace="kubeflow-example-com",
+            cluster_name=cluster,
+            region=region,
+        )
+
+    return configure_resource_fixture(
+        metadata=metadata,
+        request=request,
+        resource_details="created",
+        metadata_key=metadata_key,
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+
+@pytest.fixture(scope="class")
+def s3_bucket_with_training_data(metadata, cluster, region, request):
+    metadata_key = "s3-bucket"
+    bucket_name = "s3-" + RANDOM_PREFIX
+    bucket = S3BucketForKServe(name=bucket_name)
+
+    def on_create():
+        bucket.create()
+
+    def on_delete():
+        bucket.delete()
+
+    return configure_resource_fixture(
+        metadata=metadata,
+        request=request,
+        resource_details=bucket_name,
+        metadata_key=metadata_key,
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+
+@pytest.fixture(scope="class")
+def kserve_secret(metadata, cluster, region, request):
+    metadata_key = "aws-secret"
+
+    def on_create():
+        secret_config = load_yaml_file(SECRET_CONFIG_FILE)
+        secret_config["metadata"]["annotations"]["serving.kserve.io/s3-region"] = region
+        write_yaml_file(secret_config, SECRET_CONFIG_FILE)
+        kubectl_apply(SECRET_CONFIG_FILE)
+        # patch secret to IRSA in profile namespace
+        exec_shell(
+            'kubectl patch serviceaccount aws-sa -n kubeflow-example-com -p \'{"secrets": [{"name": "aws-secret"}]}\''
+        )
+
+    def on_delete():
+        cmd = f"kubectl delete secret aws-secret -n kubeflow-example-com".split()
+        subprocess.call(cmd)
+
+    return configure_resource_fixture(
+        metadata=metadata,
+        request=request,
+        resource_details="created",
+        metadata_key=metadata_key,
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+
+@pytest.fixture(scope="class")
+def kserve_inference_service(metadata, cluster, region, request):
```

Comment:
need to introduce dependency on the fixtures above otherwise this will run out of order

Reply:
what do you mean by this?